### PR TITLE
Location image upload on AWS

### LINKED ISSUE #8: AWS Image uploads

### PR DESCRIPTION
**Pull request by eric-ls**
fix #8 
allow businesses to upload images per location

screen shot 2016-11-13 at 2 56 28 pm

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
+config.paperclip_defaults = {
+  :storage => :s3,
+  :s3_host_name => 'REMOVE_THIS_LINE_IF_UNNECESSARY',
+  :bucket => 'S3_BUCKET_NAME'
+}
+
```
Comment (eric-ls) [inline config/environments/production.rb]: whoops, i will remove this

File: app/assets/javascripts/components/dashboard/business_dashboard.jsx [hunk 2]
```diff
@@ -1,34 +1,42 @@
...
- * @prop locations - collection (array) of locations attached to business
+ * Renders the home business dashboard view
+ * @prop business - the current business that is signed in
  */
 class BusinessDashboard extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { locations : this.props.locations };
+    this.state = {
+      business: {}
```
Comment (warrenshen) [inline app/assets/javascripts/components/dashboard/business_dashboard.jsx]: Trailing comma :)

File: app/assets/javascripts/components/locations/location_pickups.jsx [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
+
   constructor(props) {
     super(props);
   }
 
   render() {
-    let pickups = this.props.pickups.map((pickup, i) => {
-      return <div key={i}>{pickup.title}</div>
-    })
+    let pickups, num_pickups = 0;
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_pickups.jsx]: Ooh wow didn't realize you could do multi-assign in js!

File: app/assets/javascripts/components/locations/location_pickups.jsx [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
+
+    if (this.props.pickups) {
+      pickups = this.props.pickups.map((pickup, i) => {
+        return <div key={i}>{pickup.title}</div>
+      })
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_pickups.jsx]: Semicolon

File: app/assets/javascripts/components/locations/location_pickups.jsx [hunk 5]
```diff
@@ -25,5 +32,5 @@ class LocationPickups extends React.Component {
 }
 
 LocationPickups.propTypes = {
-  pickups : React.PropTypes.array.isRequired,
+  pickups : React.PropTypes.array,
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_pickups.jsx]: Should be `defaultProp` if not required.

File: app/assets/javascripts/components/locations/location_settings.jsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+ * @prop fetchLocation - callback function to retrieve location
+ */
+class LocationSettings extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      location: {},
+      uploadSuccess: false,
+    }
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_settings.jsx]: Semicolon

File: app/assets/javascripts/components/locations/location_settings.jsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ location: nextProps.location })
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_settings.jsx]: Semicolon - remind me to talk to you about setting state based on props (it's supposed to be avoided).

File: app/assets/javascripts/components/locations/location_settings.jsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+  _setFile = (e) => {
+    const files = e.target.files;
+    if (!files || !files[0]) {
+      return
+    }
+
+    const reader = new FileReader();
+    reader.onload = (file) => {
+      this.setState({ photo: file.target.result, })
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_settings.jsx]: Semicolon

File: app/assets/javascripts/components/locations/location_settings.jsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  _uploadFile = (e) => {
+    e.preventDefault();
+
+    const success = (data) => {
+      this.props.fetchLocation();
+
+      this.setState({
+        location: data,
+        uploadSuccess: true,
+      })
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_settings.jsx]: Semicolon

File: app/assets/javascripts/components/locations/location_settings.jsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    let params = { photo: this.state.photo, }
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_settings.jsx]: Same

File: app/assets/javascripts/components/locations/location_settings.jsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        </div>
+      </div>
+    )
+  }
+}
+
+LocationSettings.propTypes = {
+  location      : React.PropTypes.object.isRequired,
+  fetchLocation : React.PropTypes.func.isRequred,
+}
```
Comment (warrenshen) [inline app/assets/javascripts/components/locations/location_settings.jsx]: Same

File: app/assets/javascripts/requests/request.jsx [hunk 12]
```diff
@@ -53,7 +53,7 @@
       const request = this.initialize('PATCH', route);
       request.onreadystatechange = () => {
         if (request.readyState === XMLHttpRequest.DONE) {
-          if (request.status === 201 && resolve) {
+          if (request.status === 200 && resolve) {
```
Comment (warrenshen) [inline app/assets/javascripts/requests/request.jsx]: You really shouldn't do this but if you want.. instead you should change the response code the server is sending down

File: app/controllers/api/locations_controller.rb [hunk 13]
```diff
@@ -36,16 +36,20 @@ def update
 
   def location_params
   ##### TODO: permit the fields allowed when building a new location ######
-    params.require(:location).permit(
+    params.permit(
     :addr_name,
```
Comment (warrenshen) [inline app/controllers/api/locations_controller.rb]: Should be tabs for the symbols

File: app/assets/javascripts/components/locations/location_pickups.jsx (hunk 3 above)
Comment (eric-ls) [inline app/assets/javascripts/components/locations/location_pickups.jsx]: wait jk you can't... changed it to two lines instead

**Pull request merged**